client/signup/steps/domains/index.jsx

```diff
@@ -655,6 +655,23 @@ class DomainsStep extends React.Component {
 			backLabelText = translate( 'Back to My Sites' );
 		}
 
+		// Override Back link if source parameter is found below
+		const backUrlSourceOverrides = {
+			'business-name-generator': '/business-name-generator',
+			domains: '/domains',
+		};
+		const source = get( this.props, 'queryObject.source' );
+
+		let isExternalBackUrl;
+
+		if ( backUrlSourceOverrides[ source ] ) {
+			backUrl = backUrlSourceOverrides[ source ];
+			backLabelText = translate( 'Back' );
+
+			// Solves route conflicts between LP and calypso (ex. /domains).
+			isExternalBackUrl = true;
+		}
```

Comment:
I feel like at this point, it might be good to refactor this logic to a separate method. Whenever I see a "declare variable, then go through if/else to set the values for them", a dedicated method seems like a much cleaner approach.

```
let var1;
if ( something ) {
   var1 = "foo";
} else if ( somethingElse ) {
   var1 = "bar";
} else {
   var1 = "foobar";
}
```

vs (sloppy pseudo-code)

```
const var1 = getVar1();

getVar1() {
if ( something ) {
    return "foo";
}

if ( somethingElse ) {
    return "bar";
}

return "foobar";
}
```

Would probably need to return an object instead of a single value, but still seems worth it. And we can get rid of `let` and use `const` variables only :)

Reply:
Fair enough. Move back button related lines into `getBackUrl`.
Will keep this suggestion in mind. Thanks!